app/controllers/etds_controller.rb

```diff
@@ -21,14 +21,21 @@ def index
 
   # POST /etds
   # receives xml data from peoplesoft
-  def create
+  def create # rubocop:disable Metrics/AbcSize
     logger.info("RAW XML: #{request.raw_post}")
 
     return unless submission
 
-    @submission.update!(submission_attributes)
-    peoplesoft_actions
-    render_created
+    Submission.transaction do
+      submission.update!(submission_attributes)
+
+      peoplesoft_service.new_reader_action(readers:, reader_action_attributes:)
+      peoplesoft_service.new_registrar_action(registrar_action_attributes:)
```

Comment:
I inlined `peoplesoft_actions` and `render_created` because I think we should avoid moving important code into private methods because Rubocop is unhappy, unless we feel they're mere implementation details. Moving the code below makes it harder to see what a controller is actually doing.